src/com/boxboat/jenkins/library/git/GitConfig.groovy

```diff
@@ -50,4 +52,23 @@ class GitConfig extends BaseConfig<GitConfig> implements Serializable {
     static String replacePath(String base, String path) {
         return base.replaceFirst(/(?i)\{\{\s+path\s+\}\}/, path)
     }
+
+    def withCredentials(Map configMap = [:], Closure closure) {
```

Comment:
I think this class needs the same treatment as the Vault Classes where it uses a Params objects with overloads, so that coding style remains consistent

Reply:
Updated with params